src/vdp_scanner.py

```diff
@@ -0,0 +1,296 @@
+            logging.warning("Trying '%s' without TLS verification", domain)
+            try:
+                # Fallback to unverified TLS
+                result = self._hasher.hash_url(urlunparse(url), verify=False)
+            # If this also fails, fallback to HTTP
+            except (requests.exceptions.ConnectionError, requests.exceptions.Timeout):
+                logging.warning("Falling back to HTTP for '%s'", domain)
+                # Try connecting to the HTTP endpoint instead
+                try:
+                    result = self._hasher.hash_url(
+                        urlunparse(url._replace(scheme="http"))
+                    )
+                # If we're unable to successfully retrieve the URL for some reason
+                except Exception as err:
+                    self._log_vdp_failure(domain, err)
+        # Fallback to HTTP in case there is no HTTPS for the given domain
+        except (requests.exceptions.ConnectionError, requests.exceptions.Timeout):
+            logging.warning("Falling back to HTTP for '%s'", domain)
+            # Try connecting to the HTTP endpoint instead
+            try:
+                result = self._hasher.hash_url(urlunparse(url._replace(scheme="http")))
+            # If we're unable to successfully retrieve the URL for some reason
+            except Exception as err:
+                self._log_vdp_failure(domain, err)
+        except Exception as err:
+            self._log_vdp_failure(domain, err)
+
+        if not result:
+            return ("", False, False)
+
+        if result.status == 200:
+            return (result.visited_url, result.is_redirect, True)
+
+        return (result.visited_url, result.is_redirect, False)
+
+    def process_domain(self, domain_info: Dict[str, Any]) -> None:
+        """Process a domain entry from the DotGov CSV."""
+        # These are direct copies from current-federal.csv
+        vdp_result = self.check_for_vdp(domain_info["Domain Name"])
+
+        self.add_domain_result(
+            DomainResult(
+                domain_info["Domain Name"],
+                domain_info["Agency"],
+                domain_info["Organization"],
+                domain_info["Security Contact Email"],
+                *vdp_result,
+            )
+        )
+
+    def add_domain_result(self, result: DomainResult) -> None:
+        """Process the provided results for a domain."""
+        result_dict = {
+            "Domain": result.domain,
+            "Agency": result.agency,
+            "Organization": result.organization,
+            "Security Contact Email": result.security_contact,
+            "Visited URL": result.visited_url,
+            "Was it Redirected": result.is_redirect,
+            "VDP is Published": result.vdp_present,
+        }
+        self.domain_results.append(result_dict)
+
+        self.agency_results[result.agency]["Total Domains"] += 1
+
+        if result.security_contact and result.security_contact != "(blank)":
```

Comment:
Consider turning this magic value (`"(blank)"`) into a constant with an appropriate comment.

Reply:
Reasonable enough, updated in a739bed.